test/strings/util.jl

```diff
@@ -293,6 +293,8 @@ end
     @test replace("a", in("a") => typeof) == "Char"
     @test replace("a", ['a'] => typeof) == "Char"
 
+    @test replace("foobar", "foo"=>"bar", "bar"=>"baz") == "bazbaz"
```

Comment:
Should also test what happens when the second pair matches a sequence which has been produced by the first one.